mailpoet/assets/js/src/newsletters/send.tsx

```diff
@@ -793,84 +810,92 @@ class NewsletterSendComponent extends Component<
           automationId="newsletter_send_heading"
         />
         <ErrorBoundary>
-          <Form
-            id="mailpoet_newsletter"
-            fields={fields}
-            automationId="newsletter_send_form"
-            item={this.state.item}
-            loading={this.state.loading}
-            onChange={this.handleFormChange}
-            onSubmit={this.handleSave}
+          <SendContext.Provider
+            /* not sure how to resolve this in a class component, the suggestion is to use useMemo */
+            /* eslint-disable-next-line react/jsx-no-constructed-context-values */
+            value={{
+              saveDraftNewsletter: this.saveDraftNewsletter,
+            }}
```

Comment:
The issue is that you're creating a new object for each render. Any solution that can prevent that will be fine. I don't know much about class components, but probably sth like `this.value`, where `value = { saveDraftNewsletter: this.saveDraftNewsletter };` could do it? At least in case it's only `render` that's called again and again.

Reply:
In 58978ec921db86709636ff4cc03ebc4c6c7cdd49 I changed this so the value exists in the component's state. Does that seem reasonable to you?

Reply:
@johnolek Yeah, that looks good. Another good idea how to make the object identity stable 👍